model/canvas.cc

```diff
@@ -148,19 +148,9 @@ namespace minsky
           if (auto to=closestInPort(x,y)) {
             model->addWire(static_cast<shared_ptr<Port>&>(fromPort),to);
             fromPort.reset();
-            auto& toItem=to->item();
-            if (auto ravel=dynamic_cast<Ravel*>(&toItem))
-            {
-              auto state = ravel->getState();
-              if(state.empty()) {
-                minsky().reset();
-              }
-            } else if(auto sheet=dynamic_cast<Sheet*>(&toItem)) {
-              if(sheet->empty()) {
-                minsky().reset();
-              }
-            }
-            
+            if(!to->item().inputDefined()) {
```

Comment:
We've inverted the logic here. Perhaps if we call this method inputIsScalar instead (returning true if no input, or all connected inputs are scalar (rank 0), then call reset here if !inputIsScalar().